### PR TITLE
linux_6_8,linux_6_9: remove

### PR DESCRIPTION
**Pull request by alyssais**
## Description of changes

EOL upstream.

This seems to mean we no longer need any patches to support modern Rust versions in the kernel!

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by K900** [comment]
Eval failed :(

File: pkgs/os-specific/linux/zfs/unstable.nix [hunk 1]
```diff
@@ -17,7 +17,7 @@ callPackage ./generic.nix args {
   # check the release notes for compatible kernels
   kernelCompatible = kernel.kernelOlder "6.11";
 
-  latestCompatibleLinuxPackages = linuxKernel.packages.linux_6_9;
+  latestCompatibleLinuxPackages = linuxKernel.packages.linux_6_6;
```
Comment (Ma27) [inline pkgs/os-specific/linux/zfs/unstable.nix:20]: It says `kernelOlder "6.11"` above, so we should be able to use linux_6_10 here?

**Comment by Ma27** [comment]
Hmm, just realized we can also remove 6.8/6.9 from `hardened/patches.json`, right?

**Comment by alyssais** [comment]
depends if the update script will just put it back again

File: pkgs/os-specific/linux/zfs/unstable.nix (hunk 1 above)
Comment (alyssais) [reply pkgs/os-specific/linux/zfs/unstable.nix:20]: I think that might be wrong, because I checked the sources, and it said 6.8.

**Comment by Ma27** [comment]
> depends if the update script will just put it back again

You're right, yes it does, disregard it then.

File: pkgs/os-specific/linux/zfs/unstable.nix (hunk 1 above)
Comment (Ma27) [reply pkgs/os-specific/linux/zfs/unstable.nix:20]: Below are apparently a few 6.10 compat patches in `unstable.nix` that were approved by upstream.

Comment (alyssais) [reply pkgs/os-specific/linux/zfs/unstable.nix:20]: oh right! thanks

File: pkgs/top-level/linux-kernels.nix [hunk 2]
```diff
@@ -279,8 +257,6 @@ in {
     linux_5_15_hardened = hardenedKernelFor kernels.linux_5_15 { };
     linux_6_1_hardened = hardenedKernelFor kernels.linux_6_1 { };
     linux_6_6_hardened = hardenedKernelFor kernels.linux_6_6 { };
-    linux_6_8_hardened = hardenedKernelFor kernels.linux_6_8 { };
-    linux_6_9_hardened = hardenedKernelFor kernels.linux_6_9 { };
```
Comment (Ma27) [inline pkgs/top-level/linux-kernels.nix:283]: I'm sorry I didn't notice it earlier, but for the _hardened we also have `throws` in this file that are still missing for 6.8/6.9.

Comment (alyssais) [reply pkgs/top-level/linux-kernels.nix:283]: No worries, fixed!

**Comment by emilazy** [comment]
> depends if the update script will just put it back again

It shouldn’t, but it should also remove them automatically:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/kernel/hardened/update.py#L212-L215

**Review by Ma27 — COMMENTED** [review]
Did a sanity check, found a final reference I'm afraid:

```
pkgs/os-specific/linux/zfs/2_2.nix
20:  latestCompatibleLinuxPackages = linuxKernel.packages.linux_6_8;
```

**Comment by LovingMelody** [comment]
ZFS should be resolved with [v2.2.5](https://github.com/openzfs/zfs/releases/tag/zfs-2.2.5)

**Comment by LovingMelody** [comment]
It's merged to master with 6.10 with patches

**Comment by Ma27** [comment]
Eh I somehow had in mind that we had 6.10 there, apologies!
Will file a followup soonish.

**Pull request merged**